src/app-layer-htp.c

```diff
@@ -465,6 +465,51 @@ void AppLayerHtpNeedFileInspection(void)
     SCReturn;
 }
 
+static void AppLayerHtpSetStreamDepthFlag(void *tx, uint8_t flags)
+{
+    HtpTxUserData *tx_ud = (HtpTxUserData *) htp_tx_get_user_data((htp_tx_t *)tx);
+    if (tx_ud) {
+        if (flags & STREAM_TOCLIENT) {
+            tx_ud->tcflags |= HTP_STREAM_DEPTH_SET;
+        } else {
+            tx_ud->tsflags |= HTP_STREAM_DEPTH_SET;
+        }
+    }
+}
+
+static int AppLayerHtpCheckStreamDepth(const HtpTxUserData *tx_ud)
```

Comment:
should the direction in which we check depend on the caller, instead of the tx_ud->t(s|c)flags?

Reply:
Sorry I don't get your point here

Reply:
Ok understood.